### PR TITLE
Multiple changes, including adding travis and github keys; new README; and updated tests

### PR DESCRIPTION
**Pull request by andrie**
Add new function `add_travis_user()`; imports `httr` to read travis key

### CODE REVIEW

File: R/users.R [hunk 1]
```diff
@@ -20,13 +20,13 @@
 add_user <- function(email, public_key, vault = NULL) {
   assert_that(is_email_address(email))
   vault <- find_vault(vault)
-
+
```
Comment (gaborcsardi) [inline R/users.R]: You have whitespace errors.

File: R/users.R [hunk 2]
```diff
@@ -20,13 +20,13 @@
...
   vault <- find_vault(vault)
-
+  
   user_file <- get_user_file(vault, email)
   if (file.exists(user_file)) {
     stop("User ", sQuote(email), " already exists in this vault. ",
          "To update it, remove the old key, and add the new one.")
   }
-
+
```
Comment (gaborcsardi) [inline R/users.R]: Another whitespace error.

File: R/users.R [hunk 3]
```diff
@@ -45,6 +45,29 @@ add_github_user <- function(github_id, email = NULL, vault = NULL) {
   ## TODO
 }
 
+#' Add a user via their Travis repo
+#' 
+#' @param travis_repo Name of Travis repository, usually in a format 
+#' <<username>>/<<repo>>
```
Comment (gaborcsardi) [inline R/users.R]: Best to make this code, in backticks: `` `<<username>>/<<repo>>` ``

File: R/users.R [hunk 4]
```diff
@@ -59,19 +82,19 @@ add_github_user <- function(github_id, email = NULL, vault = NULL) {
 delete_user <- function(email, vault = NULL) {
   assert_that(is_email_address(email))
   vault <- find_vault(vault)
-
+
```
Comment (gaborcsardi) [inline R/users.R]: Whitespace error.

File: R/users.R [hunk 5]
```diff
@@ -59,19 +82,19 @@ add_github_user <- function(github_id, email = NULL, vault = NULL) {
...
   vault <- find_vault(vault)
-
+  
   ## Check if user exists
   user_file <- get_user_file(vault, email)
   if (!file.exists(user_file)) {
     stop("User ", sQuote(email), " does not exist")
   }
-
+
```
Comment (gaborcsardi) [inline R/users.R]: Whitespace error.

File: R/users.R [hunk 6]
```diff
@@ -59,19 +82,19 @@ add_github_user <- function(github_id, email = NULL, vault = NULL) {
...
   user_file <- get_user_file(vault, email)
   if (!file.exists(user_file)) {
     stop("User ", sQuote(email), " does not exist")
   }
-
+  
   ## Get all secrets they have access to
   secrets <- list_user_secrets(vault, email)
-
+
```
Comment (gaborcsardi) [inline R/users.R]: Whitespace.

File: R/users.R [hunk 7]
```diff
@@ -59,19 +82,19 @@ add_github_user <- function(github_id, email = NULL, vault = NULL) {
...
-
+  
   ## Get all secrets they have access to
   secrets <- list_user_secrets(vault, email)
-
+  
   ## Remove everything in one go. This is still not atomic, of course...
   file.remove(user_file, secrets)
-
+
```
Comment (gaborcsardi) [inline R/users.R]: Whitespace.

**Comment by gaborcsardi** [comment]
Thanks! Added some nitpicking comments. 

I'll probably also rewrite it to use `curl`, that's a bit lighter than `httr`, and we don't really need `httr` functionality here.

**Comment by andrie** [comment]
Nitpicking is good.  I'm not entirely sure what you mean by "whitespace error" - I presume you don't want the empty lines.  I have deleted those.

I agree with taking the lighter weight dependency on `curl` instead of `httr`.  Will implement.

**Comment by andrie** [comment]
Now imports from curl and jsonlite, instead of httr.
Also fixes an issue that when creating a package vault, the path is `./inst/vault`, but reading from an installed package, the path is `./vault`

**Comment by andrie** [comment]
Added functionality for `add_github_user()`

**Comment by andrie** [comment]
This last pull request adds a small bug fix, as well as new tests for `add_travis_user()` and `add_github_user()`.

I also renamed the keys in `./inst/user_keys` to `alice.pub`, `bob.pub`, etc. so the code is easier to read and the concepts easier to explain in a vignette and readme.

**Comment by andrie** [comment]
I rewrote the `README` using `rmarkdown`, i.e. now there is a `README.Rmd` file

**Comment by andrie** [comment]
I have refactored the tests into multiple files, and added additional tests for `update_secret()`, `share_secret()` and `local_key()`.  I also added code coverage into `.travis.yml` and added the code coverage badge.  Code coverage now is greater than 95%.

**Comment by gaborcsardi** [comment]
Great, thanks!

**Pull request merged**